tools/calyx-pass/src/util.rs

```diff
@@ -0,0 +1,30 @@
+use std::process::Command;
+
+/// Returns the standard output from executing a given command `cmd` with
+/// arguments `args`. Fails when the command fails or succeeds but with a
+/// nonzero exit code (and `wants_zero`).
+pub fn capture_command_stdout(
+    cmd: &str,
+    args: &[&str],
+    wants_zero: bool,
+) -> std::io::Result<String> {
+    let output = Command::new(cmd).args(args).output()?;
+    if !output.status.success() && wants_zero {
+        Err(std::io::Error::new(
+            std::io::ErrorKind::Other,
+            format!(
+                "Command `{} {}` did not execute successfully (code={}). Stderr: {}",
+                cmd,
+                args.iter()
+                    .map(|str| str.to_string())
+                    .collect::<Vec<_>>()
```

Comment:
`collect_vec` instead

Reply:
THANK YOU

Reply:
I wasn't able to figure out how to use this function on a `Map` iterator.